### DISCUSSION TITLE
read an external file

### DISCUSSION THREAD

**Discussion by eduardogrodriguez89** [idaholab/moose#24427] [Q&A General]
Hello!
I want to read some materials properties from an external file (.txt, .cvs), like young_modulus and poisson_ratio, and add them to the material modulus.  Is it possible? 
And a second question, if I have a file with strain-stress information, (this could be a plot of a strain-stress relation for a material) 
Can a read a specific strain and take the corresponding stress?

**Comment by GiudGiud** [comment]
Hello

It's definitely possible.
You ll want to use this object for computing the elasticity tensor from variable (material properties) young modulus and poisson ratio
https://mooseframework.inl.gov/source/materials/ComputeVariableIsotropicElasticityTensor.html

To read material properties from CSV, we need more details. How are they tabulated in that file? As a function of space? Temperature?

I dont think we can do the second thing. @hugary1995 will know for sure

Guillaume

**Comment by eduardogrodriguez89** [comment]
I will read it as a function of space

**Reply by GiudGiud** [reply]
You can load CSVs for space-dependent profiles using this object:
https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html
This will create a function. Then you can use this object to make it (them) materials:
https://mooseframework.inl.gov/source/materials/GenericFunctionMaterial.html
